TFNetworkLayer.py

```diff
@@ -1107,53 +1111,218 @@ def batch_norm(self, data,
       tf.nn.batch_normalization()
       https://github.com/deepmind/sonnet/blob/master/sonnet/python/modules/batch_norm.py
     """
+    from operator import mul
+    from functools import reduce
+
     with tf.variable_scope("batch_norm"):
-      if masked_time:
-        x = data.get_placeholder_flattened(keep_dims=True)
-        mean, variance = tf.nn.moments(x, axes=[0], keep_dims=True)
-      else:
-        x = data.placeholder
-        mean, variance = tf.nn.moments(x, axes=data.get_axes(exclude_feature=True), keep_dims=True)
+      # Creating variables
+      fused = fused and not force_sample
```

Comment:
Why?
Also, don't write code like this. Writing it like this is much easier to read and understand:
```
if force_sample:
  fused = False
```